Sources/Identity/CustomerInfoManager.swift

```
@@ -183,12 +217,13 @@ class CustomerInfoManager {
                     self.fetchAndCacheCustomerInfo(appUserID: appUserID,
                                                    isAppBackgrounded: isAppBackgrounded,
                                                    completion: completion)
+                    // TODO: Track getCustomerInfoResult if !completionCalled
                 }
             }
         }
     }
 
-    func cachedCustomerInfo(appUserID: String) -> CustomerInfo? {
+    func cachedCustomerInfo(appUserID: String) throws -> CustomerInfo? {
```

I've changed this method to be throwing so that diagnostics event can gather more information about why fetching from the cache didn't work. 

I'm not sure if this is worth it instead of sending the same `BackendError.missingCachedCustomerInfo()` that we send to callers, as this will only be useful when using the `.fromCacheOnly` cache policy (as the other policies using the cache will fall back to fetching the latest one in case of error).

---

Hmm I see we were swallowing any exception before... I think it's fine to expose that to get more details of any failures yeah... I think this is ok!

---

But to be clear, even with this method now `throwing`, we keep swallowing these errors in the SDK and the behavior for developer calls that go through this method won't change. 
You can see it [here](https://github.com/RevenueCat/purchases-ios/blob/09279498f4591b9fdecbb605a75837a6c71f300c/Sources/Identity/CustomerInfoManager.swift#L125)
```swift
     self.operationDispatcher.dispatchOnMainActor {
                let result = Result { try self.cachedCustomerInfo(appUserID: appUserID) }

                // We want the specific error for diagnostics
                let resultForDiagnostics = Result(result.value as? CustomerInfo,
                                                  result.error ?? BackendError.missingCachedCustomerInfo())
                self.trackGetCustomerInfoResultIfNeeded(trackDiagnostics: trackDiagnostics,
                                                        startTime: startTime,
                                                        cacheFetchPolicy: fetchPolicy,
                                                        hadUnsyncedPurchasesBefore: nil,
                                                        usedOfflineEntitlements: false,
                                                        result: resultForDiagnostics)

                // But for callers we only pass `.missingCachedCustomerInfo()` error
                completion?(
                    Result(result.value as? CustomerInfo, .missingCachedCustomerInfo())
                )
```
where we only used the newly-thrown error for diagnostics (`resultForDiagnostics`) but for customers we keep the same behavior of throwing the `.missingCachedCustomerInfo()` error in case of `nil` `CustomerInfo`.

Does it make sense?

---

Yup yup! We are now only "processing" errors for diagnostics, but otherwise keep swallowing those errors. Makes sense for this PR 👍